src/Adapter/Supplier/QueryHandler/GetSupplierForViewingHandler.php

```diff
@@ -118,77 +117,116 @@ private function getSupplierProducts(Supplier $supplier, LanguageId $languageId)
         $products = [];
         $supplierProducts = $supplier->getProductsLite($languageId->getValue());
 
-        try {
-            foreach ($supplierProducts as $productData) {
-                $product = new Product($productData['id_product'], false, $languageId->getValue());
-                $product->loadStockData();
-
-                $productCombinations = $product->getAttributeCombinations($languageId->getValue());
-                $combinations = [];
-
-                foreach ($productCombinations as $combination) {
-                    $attributeId = $combination['id_product_attribute'];
-                    if (!isset($combinations[$attributeId])) {
-                        $productInfo = Supplier::getProductInformationsBySupplier(
-                            $supplier->id,
-                            $product->id,
-                            $combination['id_product_attribute']
-                        );
-                        $isoCode = Currency::getIsoCodeById((int) $productInfo['id_currency'])
-                            ?: $this->defaultCurrencyIsoCode;
-                        $formattedWholesalePrice = null !== $productInfo['product_supplier_price_te']
-                            ? $this->locale->formatPrice($productInfo['product_supplier_price_te'], $isoCode)
-                            : null;
-                        $combinations[$attributeId] = [
-                            'reference' => $combination['reference'],
-                            'supplier_reference' => $combination['supplier_reference'],
-                            'wholesale_price' => $formattedWholesalePrice,
-                            'ean13' => $combination['ean13'],
-                            'upc' => $combination['upc'],
-                            'quantity' => $combination['quantity'],
-                            'attributes' => '',
-                        ];
-                    }
-                    $attribute = sprintf(
-                        '%s - %s',
-                        $combination['group_name'],
-                        $combination['attribute_name']
-                    );
-
-                    if (!empty($combinations[$attributeId]['attributes'])) {
-                        $attribute = sprintf(', %s', $attribute);
-                    }
-
-                    $combinations[$attributeId]['attributes'] = $attribute;
-                }
+        foreach ($supplierProducts as $productData) {
+            $product = new Product($productData['id_product'], false, $languageId->getValue());
+            $product->loadStockData();
+            $combinations = $this->findProductSupplierCombinations($product, $supplier, $languageId);
+
+            if (empty($combinations)) {
+                $products[] = $this->buildNonCombinationSupplierProduct($product, $supplier);
+                continue;
+            }
+
+            $products[] = [
```

Comment:
the `$products` array entries does not have the same keys. it's disturbing for me :D

Reply:
Ow I see. It's the combinations which have the extra keys like ean13 or quantity

Reply:
Yes every products are in the same array, standard product have all the details But for products with combinations the details are only provided for each combinations
This code could be improved later with objects instead of flat array but it's not the purpose of this PR